src/redux/actions/posts.js

```diff
@@ -0,0 +1,27 @@
+import { database } from 'firebase'
+
+export const getPostsRequested = () => dispatch => {
+  dispatch({ type: 'GET_POSTS_REQUESTED' })
+}
+
+export const getPostsRejected = () => dispatch => {
+  dispatch({ type: 'GET_POSTS_REJECTED', payload: 'An error occurred when retrieving data' })
+}
+
+export const getPostsSuccess = post => dispatch => {
+  dispatch({ type: 'GET_POSTS_SUCCESS', payload: post })
+}
+
+export const getPosts = () => dispatch => {
+  dispatch(getPostsRequested())
+  return database()
+    .ref('/posts/')
+    .once('value', snap => {
```

Comment:
your call to `.once` isn't quite right. You've used a combination of callbacks: 
`snap => {` and promises: `.catch`.

You can use either, but not both together! (Maybe you can use both together, but probably shouldn't!)

More details of how it works: 
https://firebase.googleblog.com/2016/01/keeping-our-promises-and-callbacks_76.html